### PR TITLE
Fix unknown environment variable for `DD_PROCESS_AGENT_ENABLED`, `DD_PROCESS_CONFIG_ENABLED`, and `DD_SYSTEM_PROBE_ENABLED`

### PR DESCRIPTION
**Pull request by just-chillin**
### What does this PR do?

Removes the `Unknown environment variable` message when using `DD_PROCESS_AGENT_ENABLED` or `DD_PROCESS_CONFIG_ENABLED` environment variables. This fix also prevents the same thing from happening with system-probe environment variables.

### Motivation

This fix is meant to address a bug introduced by https://github.com/DataDog/datadog-agent/pull/9646 and https://github.com/DataDog/datadog-agent/pull/10097

### Additional Notes

The system-probe error occurs because of a change made in https://github.com/DataDog/datadog-agent/pull/10097/files, where the global config is loaded before the system probe is merged. The solution to this is to move the system probe merging outside of `NewAgentConfig` and put it before `LoadConfigIfExists` is called.

### Describe how to test/QA your changes

Run the process agent with these three environment variables:
- `DD_PROCESS_AGENT_ENABLED`
- `DD_PROCESS_CONFIG_ENABLED`
- `DD_SYSTEM_PROBE_ENABLED`

Make sure that no `Unknown environment variable` warning is shown in the logs.

To make sure the flare still works, you should run both the agent and the process-agent with the `DD_PROCESS_CONFIG_ENABLED=true` or `DD_PROCESS_AGENT_ENABLED=true` environment variables still set. Next you should collect a flare using `datadog-agent flare`. In the zip file there should be a file called `envvars.log`, in this file check to make sure that either `DD_PROCESS_CONFIG_ENABLED` or `DD_PROCESS_AGENT_ENABLED` is shown.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by djmitche — COMMENTED** [review]
This fix looks good, but I'm confused about process_*agent*.enabled and process_*config*.enabled.  It looks like before this patch, `DD_PROCESS_AGENT_ENABLED=true` led to `process_config.enabled` being true.  But after, only `DD_PROCESS_CONFIG_ENABLED=true` would do so.  Is that intentional?

File: pkg/flare/envvars.go [hunk 1]
```diff
@@ -71,7 +71,6 @@ var allowedEnvvarNames = []string{
 	"DD_APM_PROFILING_ADDITIONAL_ENDPOINTS",
 
 	// Process agent
-	"DD_PROCESS_AGENT_ENABLED",
```
[the remark this answers is not in the document]
Comment (xornivore) [reply pkg/flare/envvars.go:74]: Thank you for the explanation, mind adding some testing steps for this change in the PR description? 🙇

**Comment by just-chillin** [comment]
> This fix looks good, but I'm confused about process__agent_.enabled and process__config_.enabled. It looks like before this patch, `DD_PROCESS_AGENT_ENABLED=true` led to `process_config.enabled` being true. But after, only `DD_PROCESS_CONFIG_ENABLED=true` would do so. Is that intentional?

Hey, thank you for the comment! 👋 

Our team needs to support both `DD_PROCESS_AGENT_` and `DD_PROCESS_CONFIG_` prefixes. So instead of using `config.BindEnvAndSetDefault` directly, we use our own helper called `procBindEnvAndSetDefault`, which defines the variable for both prefixes. So as long as we are using that helper this change should still make both environment variables valid.

**Review by djmitche — APPROVED** [review]
Oh, I missed the wrapper -- thanks!

File: pkg/flare/envvars.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (just-chillin) [reply pkg/flare/envvars.go:74]: Sure, no problem. I added some info about how to test the flare in the description.

**Comment by just-chillin** [comment]
I added a fix for system-probe environment variables, and updated the testing steps.

File: pkg/process/config/config_test.go [hunk 2]
```diff
@@ -12,6 +12,7 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	sysconfig "github.com/DataDog/datadog-agent/cmd/system-probe/config"
```
Comment (xornivore) [inline pkg/process/config/config_test.go:15]: Mind splitting into a separate import section?

**Review by xornivore — APPROVED** [review]
👍 Mind updating the "What does this PR do" section with a mention of `DD_SYSTEM_PROBE_ENABLED`?

File: pkg/process/config/config_test.go (hunk 2 above)
Comment (just-chillin) [reply pkg/process/config/config_test.go:15]: Oh thanks, I missed that one

**Pull request merged**